_pytest/python.py

```diff
@@ -1246,6 +1260,11 @@ def __exit__(self, *tp):
         suppress_exception = issubclass(self.excinfo.type, self.expected_exception)
         if sys.version_info[0] == 2 and suppress_exception:
             sys.exc_clear()
+        if self.match_expr:
+            try:
```

Comment:
I think just calling `self.excinfo.match` should be enough, as this will produce the same behavior as:

```python
with pytest.raises(...) as excinfo:
    func()
excinfo.match(text)
```

Or am I missing something?

I guess one issue in that case is that the traceback will point to pytest's internals, but to prevent that we just need to declare a `__tracebackhide__` variable:

```python
__tracebackhide__ = True
if self.match_expr:
    self.excinfo.match(self.match_expr)
```